File: example/dart/class/accessor.dart
class Employee {
  String fullName;
}

const passcode = 'secret passcode';

class EmployeeOptimize {
  String _fullName;

  get fullName {
    return this._fullName;
  }

  set fullName(String newName) {
    if (passcode == 'secret passcode') {
      this._fullName = newName;
    } else {
      print('Error: Unauthorized update of employee!');
    }
  }
}

main(List<String> args) {
  var employee = new EmployeeOptimize();
  employee.fullName = 'Crixus Shen';
  print('${employee.fullName}');
}

File: example/typescript/class/accessor.ts
class Employee {
  fullName: string
}

const passcode = 'secret passcode'

class EmployeeOptimize {
  private _fullName: string

  get fullName(): string {
    return this._fullName
  }

  set fullName(newName: string) {
    if (passcode == 'secret passcode') {
      this._fullName = newName
    } else {
      console.error('Error: Unauthorized update of employee!')
    }
  }
}

var employee = new EmployeeOptimize()
employee.fullName = 'Crixus Shen'
console.log(employee.fullName)